graphql/apollo.ts

```diff
@@ -28,7 +28,28 @@ export default function getApolloClient ( serverSideContext?: GetServerSideProps
 	// Server-side / static: Return a new instance every time.
 	if ( isServerSide ) {
 		// @ts-expect-error: Express locals are not defined on Next.js request.
-		const { requestContext } = serverSideContext?.res?.locals || {};
+		let { requestContext } = serverSideContext?.res?.locals || {};
+
+		// @ts-expect-error: res may not be defined
+		const sourceId = serverSideContext?.res?.getHeader('x-source-id');
+		// @ts-expect-error: res may not be defined
+		const pathName = serverSideContext?.res?.getHeader('x-request-path');
+
+		// Add the requestID for the original API/Page request to each WP query's context
+		if (sourceId) {
+			requestContext = {
+				...requestContext,
+				sourceId
+			};
+		}
+
+		// Add the path name for the original API/Page request to each WP query's context
+		if (pathName) {
```

Comment:
I think it would be better to assign something consistent rather than conditionally. Like:

```
const sourceId = /* ... */ || null;
const pathName = /* ... */ || null;

serverSideContext?.res.locals = {
  ...serverSideContext?.res.locals,
  sourceId,
  pathName,
};
```

Reply:
I tweaked it so that requestContext is consistent regardless of what values are present in sourceId and pathName. Now, for something like a static page where these are not set they will show up as null vs a regular page where they will be populated. I couldn't get it simpler because conditional operators aren't allowed for assignments, unfortunately. Let me know what you think!